tools/topology2/include/pipelines/pipeline-volume-capture.conf

```diff
@@ -0,0 +1,162 @@
+#
+# Volume capture pipeline
+#
+# A simple pipeline. All attributes defined herein are namespaced by alsatplg to
+# "pipeline-volume-capture.attribute_name"
+#
+# Usage: this component can be used by declaring in the top-level topology conf file as follows:
+#
+# 	Object.pipeline-volume-capture."N" {
+#		pipeline_id	X
+#		pcm_id		Y
+#		pcm_name	"Headset"
+#		format		"s16le"
+#		period		1000
+#		time_domain	"timer"
+#		channels	2
+#		rate		48000
+#	}
+#
+# Where X and Y are unique integers for pipeline ID and PCM ID in the parent object.
+# and N is the unique instance number for this pipeline object within the same alsaconf node.
+#
+<include/components/volume.conf>
+<include/components/buffer.conf>
+<include/components/host.conf>
+<include/components/pipeline.conf>
+<include/common/pipe_endpoint.conf>
+<include/common/source.conf>
+<include/common/sink.conf>
+<include/common/pcm.conf>
+<include/common/pcm_caps.conf>
+
+#
+#
+# (sink) host.N.playback <- buffer.N.0 <- volume.N.0 <- buffer.N.1 (source endpoint)
+#
+Class.Pipeline."pipeline-volume-capture" {
+
+	DefineArgument."pipeline_id" {
+		type "integer"
+	}
+
+	DefineArgument."pcm_id" {
+		type "integer"
+	}
+
+	<include/pipelines/pipeline-common.conf>
+
+	# Controls names for the PGA widget's mixers
+	DefineAttribute."volume_mixer_name" {}
+	DefineAttribute."switch_mixer_name" {}
+
+	attributes {
+		mandatory [
+			"channels"
+			"format"
+			"rate"
+			"direction"
+			"dynamic"
+			"period"
+			"time_domain"
+			"pcm_name"
+			"volume_ctl_name"
+		]
+		immutable [
+			"direction"
+		]
+	}
+
+	Object.pipeline."0" {
+		core			0
+		frames			0
+		priority		0
+		mips			5000
+	}
+
+	# Pipeline objects
+	Object.host."0" {
+		period_sink_count	0
+		period_source_count	2
+		widget_type		"aif_out"
+	}
+
+	Object.buffer."0" {
+		index		0
```

Comment:
btw, can we take index from the index in the name declaration ?

Reply:
Yes, I should merge these two.

Reply:
done